src/app.ts

```diff
@@ -13,14 +13,127 @@ interface TemperatureSummary {
   average: number
 }
 
+/**
+ * Temperatures is a map that stores cities
+ */
+interface Temperatures {
+  [city: string]: City
+}
+/**
+ * City is an object that describes the city's
+ * counterReadings, historical high, low, average
+ * and stores Dates
+ */
+interface City {
+  countedReadings: number
+  histHigh: number
+  histLow: number
+  histAvg: number
+  dates: Dates
+}
+/**
+ * Dates is a map where the index is a date
+ * that stores an array of temperatures and a
+ * summary object that describes that date.
+ */
+interface Dates {
+  [date: string]: {
+    temperature: number[]
+    summary: TemperatureSummary
+  }
+}
+
+const temperatures: Temperatures = {}
+
+/**
+ * processReadings will iterate from the given array and
+ * store the city, time and temp inside the global Temperatures
+ * map of maps.
+ * If the city or time does not exist, it will initialize
+ * an empty object with empty values
+ * If it does exist, it will push the temperature in the array
+ * and update internal statistic values for the city
+ */
 export function processReadings(readings: TemperatureReading[]): void {
   // add here your code
-}
+  readings.forEach((reading) => {
+    const city = reading.city
+    const date = reading.time.toDateString()
+    const tmp = reading.temperature
```

Comment:
Not completely necessary and definitely a preference thing. But destructuring here might be nice.

```suggestion
    const { city, date, temperature: tmp } = reading
```

Reply:
Also technically you don't need to call `toDateString` on the date. If you use it as a key or console log it, it will automatically be a string. You'll get a different result between the browser and node though. It may be safer to use the time since epoch as the key instead? That may be easier to work with than a time stamped string?

```
read.time.getTime()
```